src/hooks/useTransition/index.ts

```diff
@@ -71,7 +71,11 @@ export const useTransition = ({
   }, [onExiting, handleExited, timeout])
 
   useIsomorphicLayoutEffect(() => {
-    if (transitioning) return
+    if (state === TransitionState.EXITING) return
```

Comment:
**suggestion:** We should have the state in the dependency array, otherwise we might be using an old value😅 Hope it still works with it synced, as it does with transitioning that also depends on this variable.